extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1021,17 +1091,8 @@ export class AmpA4A extends AMP.BaseElement {
 
     this.isCollapsed_ = false;
 
-    // Allow embed to release its resources.
-    if (this.friendlyIframeEmbed_) {
-      this.friendlyIframeEmbed_.destroy();
-      this.friendlyIframeEmbed_ = null;
-    }
-
     // Remove rendering frame, if it exists.
-    if (this.iframe && this.iframe.parentElement) {
-      this.iframe.parentElement.removeChild(this.iframe);
-      this.iframe = null;
-    }
+    this.destroyFrame();
```

Comment:
Yeah this is somewhat complex due to the fact that we only want to destroy the frame if its not AMP (as we do not want to allow JS to execute on pages no longer visible to the user).  So for AMP creatives when unlayoutCallback occurs, we should leave the frame in place and cancel any pending refresh promises (which I believe would happen anyway given we increment the promiseId though you need to add the check as I mentioned in a previous comment).  However, we want to ensure that if the user swipes back to the page, the refresh interval either continues or restarts depending on production definition.